src/daisy_seed.cpp

```diff
@@ -245,23 +235,53 @@ void DaisySeed::ConfigureAudio()
     sai_config.bit_depth       = SaiHandle::Config::BitDepth::SAI_24BIT;
     sai_config.a_sync          = SaiHandle::Config::Sync::MASTER;
     sai_config.b_sync          = SaiHandle::Config::Sync::SLAVE;
-    sai_config.a_dir           = SaiHandle::Config::Direction::TRANSMIT;
-    sai_config.b_dir           = SaiHandle::Config::Direction::RECEIVE;
     sai_config.pin_config.fs   = {DSY_GPIOE, 4};
     sai_config.pin_config.mclk = {DSY_GPIOE, 2};
     sai_config.pin_config.sck  = {DSY_GPIOE, 5};
-    sai_config.pin_config.sa   = {DSY_GPIOE, 6};
-    sai_config.pin_config.sb   = {DSY_GPIOE, 3};
+
+    // Device-based Init
+    switch(CheckBoardVersion())
+    {
+        case DAISY_SEED_1_1:
+        {
+            // Data Line Directions
+            sai_config.a_dir         = SaiHandle::Config::Direction::RECEIVE;
+            sai_config.pin_config.sa = {DSY_GPIOE, 6};
+            sai_config.b_dir         = SaiHandle::Config::Direction::TRANSMIT;
+            sai_config.pin_config.sb = {DSY_GPIOE, 3};
+            I2CHandle::Config i2c_config;
+            i2c_config.mode           = I2CHandle::Config::Mode::I2C_MASTER;
+            i2c_config.periph         = I2CHandle::Config::Peripheral::I2C_2;
+            i2c_config.speed          = I2CHandle::Config::Speed::I2C_400KHZ;
+            i2c_config.pin_config.scl = {DSY_GPIOH, 4};
+            i2c_config.pin_config.sda = {DSY_GPIOB, 11};
+            I2CHandle i2c_handle;
+            i2c_handle.Init(i2c_config);
+            Wm8731::Config codec_cfg;
+            codec_cfg.Defaults();
+            Wm8731 codec;
```

Comment:
that's a temporary object on the stack. Do we really not need this object anymore after the initial `codec.Init()`? Just being cautious here

Reply:
In terms of compatibility with the original Daisy Seed, there aren't any additional functions that were also possible with the 4556. And since `Init` is the only function the current WM8731 driver can do, I don't see a point of keeping it around once its initialized.

That said, I have other boards where I have it as a class member so that when we add more control functionality to the driver (level/filter adjustment, etc.) it can be accessed through the board class.  So I could go either way on this, but in the interest of keeping the Seed -> Seed1.1 transition as seamless as possible, I'm leaning on the side of it not being kept.

Reply:
okay, if its not needed after the initial setup, then it can be safely destroyed afterwards